docs/app/Examples/elements/List/Types/ListDivided.js

```diff
@@ -0,0 +1,32 @@
+import React from 'react'
+import { List } from 'stardust'
+
+const ListDivided = () => {
+  return (
+    <List divided relaxed>
+      <List.Item>
+        <List.Icon name='github' size='large' verticalAlign='middle' />
+        <List.Content>
+          <List.Header as='a'>Semantic-Org/Semantic-UI</List.Header>
+          <List.Description as='a'>Updated 10 mins ago</List.Description>
+        </List.Content>
+      </List.Item>
+      <List.Item>
+        <List.Icon name='github' size='large' verticalAlign='middle' />
+        <List.Content>
+          <List.Header as='a'>Semantic-Org/Semantic-UI-Docs</List.Header>
+          <List.Description as='a'>Updated 22 mins ago</List.Description>
+        </List.Content>
+      </List.Item>
+      <List.Item>
+        <List.Icon name='github' size='large' verticalAlign='middle' />
```

Comment:
I think `List.Icon` might actually be the way to go. It would just be a wrapper around `Icon` that handles the `verticalAlign` prop:

```js
function ListIcon(props) {
  const { verticalAlign, className } = props
  const classes = cx(
    useValueAndKey(verticalAlign, 'verticalAlign'),
    className
  )
  const rest = getUnhandledProps(ListIcon, props)
  return <Icon {...rest} className={classes} />
}
```

In all likelihood, if I were using a `List` I'd either be:
- using the `icon` prop as a shorthand, so it wouldn't matter whether it's `Icon` or `List.Icon` under the hood.
- looking at the example page for `List` to see how to structure it and would see the `List.Icon` subcomponent there.

I could also see just building it into the `Icon` class with a comment noting that it's a specific use-case, although I really wish there were at least one other use-case for that.